### PR TITLE
fsharp template updates

### PR DESCRIPTION
**Pull request by sylvanc**
F# template updates for @soninaren

### CODE REVIEW

File: Templates/ExternalFileTrigger-FSharp/metadata.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "name": ExternalFileTrigger",
+    "name": "ExternalFileTrigger",
     "description": "$SaasFileTriggerFSharp_description",
```
Comment (mathewc) [inline Templates/ExternalFileTrigger-FSharp/metadata.json]: @soninaren Can we update these $Saas resx prefixes to match the new naming to reduce confusion?

File: Templates/GitHubCommenter-FSharp/run.fsx [hunk 2]
```diff
@@ -27,24 +29,61 @@ let SendGitHubRequest (url: string) requestBody =
...
+            | [_] -> true
+            | _::tl -> hasProp tl (x.Value<JObject>())
+            | [] -> false
+
+let rec prop<'T> (key: string list) (def: 'T) (from: JObject) =
+    match from with
+    | null -> def
+    | _ ->
+        let x = from.[key.Head]
+        match x with
```
Comment (mathewc) [inline Templates/GitHubCommenter-FSharp/run.fsx]: Holy cow - I'm not an F# programmer, but this code looks very complicated :)

File: Templates/HttpPUT(CRUD)-FSharp/run.fsx [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
-        let response = new HttpResponseMessage(HttpStatusCode.BadRequest)
-        response.Content <-
-            new StringContent("A non-empty Name must be specified.")
-        response
-    else
-        log.Info(sprintf "PersonName=%s" person.Name)
-        let updateOperation = TableOperation.InsertOrReplace(person)
-        let result = outTable.Execute(updateOperation)
-        new HttpResponseMessage(enum<HttpStatusCode>(result.HttpStatusCode))
+let Run(req: HttpRequestMessage, outTable: CloudTable, log: TraceWriter) =
```
Comment (mathewc) [inline Templates/HttpPUT(CRUD)-FSharp/run.fsx]: Why the move away from POCOs to pulling properties from a raw request?

File: Templates/HttpTrigger-FSharp/run.fsx [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
+type Named = {
+    name: string
+}
 
 let Run(req: HttpRequestMessage, log: TraceWriter) =
-    log.Info(sprintf 
-        "F# HTTP trigger function processed a request.")
+    async {
+        log.Info(sprintf 
+            "F# HTTP trigger function processed a request.")
```
Comment (mathewc) [inline Templates/HttpTrigger-FSharp/run.fsx]: Not sure how useful these log messages are, though they're probably the same for other languages @soninaren ? If we're writing a log message, we should output some additional details from the request, or remove the messages. Thoughts?

File: Templates/SasToken-FSharp/run.fsx [hunk 5]
```diff
@@ -43,28 +45,35 @@ let GetBlobSasToken (container: CloudBlobContainer) blobName permissions =
 
 let Run(req: HttpRequestMessage, log: TraceWriter) =
     async {
-        let! data = req.Content.ReadAsAsync<obj>() |> Async.AwaitTask
-        let container = data?container
+        let! data = req.Content.ReadAsStringAsync() |> Async.AwaitTask
+        let json = JObject.Parse(data)
+        let containerJ = json.["container"]
```
Comment (mathewc) [inline Templates/SasToken-FSharp/run.fsx]: containerJ seems like a strange variable name. Is this natural in F#?

File: Templates/ExternalFileTrigger-FSharp/metadata.json (hunk 1 above)
Comment (sylvanc) [inline Templates/ExternalFileTrigger-FSharp/metadata.json]: Those name changes were made in a different PR - I'd rather not mix solution file maintenance with this one :)

File: Templates/GitHubCommenter-FSharp/run.fsx (hunk 2 above)
Comment (sylvanc) [inline Templates/GitHubCommenter-FSharp/run.fsx]: Normally this would be done with `FSharp.Interop.Dynamic`, but unfortunately it fails to run on Azure Functions. When that's solved, this can be (drastically) simplified.

File: Templates/HttpTrigger-FSharp/run.fsx (hunk 4 above)
Comment (sylvanc) [inline Templates/HttpTrigger-FSharp/run.fsx]: If so, I would rather that were done as a separate PR across languages.

File: Templates/HttpPUT(CRUD)-FSharp/run.fsx (hunk 3 above)
Comment (mathewc) [inline Templates/HttpPUT(CRUD)-FSharp/run.fsx]: Why does POCO deserialization no longer work? It should - do we have a bug? Have you tried on the latest bits?

Comment (sylvanc) [inline Templates/HttpPUT(CRUD)-FSharp/run.fsx]: Using ~1 on the portal. As far as I can tell (i.e. with manual testing, so it's always possible I've messed it up!), it works with blobs but not with anything that deserialises an HttpRequestMessage body.

**Comment by soninaren** [comment]
merging this, we can push incremental updates later for a better experience.
